src/parser/ts/index.ts

```diff
@@ -0,0 +1,528 @@
+import {
+  DataQueryRequestItem,
+  TimeSeriesResponseItem,
+  QueryTarget,
+  IdEither,
+} from '../../types';
+import _, { isArray, isObjectLike, uniqBy } from 'lodash';
+import { Parser, Grammar } from 'nearley';
+import grammar from './grammar';
+import { getTimeseries, getLabelWithInjectedProps } from '../../cdfDatasource';
+import { Connector } from '../../connector';
+import { FilterType } from '../types';
+import { applyFilters } from '../../utils';
+import { parseWith } from '../events-assets/index';
+import getFilterDeep from 'deepdash/getFilterDeep';
+
+const filterDeep = getFilterDeep(_);
+const compiledGrammar = Grammar.fromCompiled(grammar);
+
+export const formQueriesForExpression = async (
+  expr: string,
+  target: QueryTarget,
+  connector: Connector
+): Promise<DataQueryRequestItem[]> => {
+  if (isSimpleSyntheticExpression(expr)) {
```

Comment:
can we get rid of this? cause here parser is run twice – inside `isSimpleSyntheticExpression ()` and at line 28. Mb we can just replace with `!getServerFilters(parsed).length;`

Reply:
ffixed